### PR TITLE
Fix #312 by running `go-mod-tidy`

### LINKED ISSUE #312: Emitted events from server reaches client after 10 to 50 seconds later
Here's the client code

```javascript
<!doctype html>
<html>
<head>
	<title>Socket.IO chat</title>
	<style>
		* { margin: 0; padding: 0; box-sizing: border-box; }
		body { font: 13px Helvetica, Arial; }
		form { background: #000; padding: 3px; position: fixed; bottom: 0; width: 100%; }
		form input { border: 0; padding: 10px; width: 90%; margin-right: .5%; }
		form button { width: 9%; background: rgb(130, 224, 255); border: none; padding: 10px; }
		#messages { list-style-type: none; margin: 0; padding: 0; }
		#messages li { padding: 5px 10px; }
		#messages li:nth-child(odd) { background: #eee; }
	</style>
</head>
<body>
	<div class="chat">
		<ul>
			<li v-for="message in messages">{{ message }}</li>
		</ul>
	</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/2.3.0/socket.io.js"></script>
<script src="https://code.jquery.com/jquery-1.11.1.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/vue/2.6.10/vue.js"></script>
</body>
<script>
    var socket = io();

    socket.on('reply', function (d) {
        console.log(d);
    });

	new Vue({
		el: '.chat',
		data: {
		    messages: []
		},
		mounted: function () {
            socket.on('reply', function (d) {
				this.messages.push(d)
            }.bind(this));
        }
	});

	console.log("msg");
</script>
</html>
```

Server code

```go
package main

import (
	"fmt"
	"github.com/googollee/go-socket.io"
	"github.com/labstack/echo/v4"
	"log"
)

func main() {
	server, err := socketio.NewServer(nil)
	if err != nil {
		log.Fatal(err)
	}
	
	app := echo.New()
	
	app.Static("/", ".")
	
	app.Any("/socket.io/", func(c echo.Context) error {
		server.ServeHTTP(c.Response(), c.Request())
		
		return nil
	})
	
	server.OnConnect("/", func(s socketio.Conn) error {
		s.SetContext("")
		fmt.Println("connected:", s.ID())
		
		s.Emit("reply", "Connected")
		return nil
	})
	
	server.OnError("/", func(e error) {
		fmt.Println("meet error:", e)
	})
	
	server.OnDisconnect("/", func(s socketio.Conn, reason string) {
		fmt.Println("closed", reason)
	})
	go server.Serve()
	defer server.Close()
	
	app.Start(":3001")
}
```

But if I create another io on server side, it works fine. Like this one

```javascript
<!doctype html>
<html>
<head>
	<title>Socket.IO chat</title>
	<style>
		* { margin: 0; padding: 0; box-sizing: border-box; }
		body { font: 13px Helvetica, Arial; }
		form { background: #000; padding: 3px; position: fixed; bottom: 0; width: 100%; }
		form input { border: 0; padding: 10px; width: 90%; margin-right: .5%; }
		form button { width: 9%; background: rgb(130, 224, 255); border: none; padding: 10px; }
		#messages { list-style-type: none; margin: 0; padding: 0; }
		#messages li { padding: 5px 10px; }
		#messages li:nth-child(odd) { background: #eee; }
	</style>
</head>
<body>
	<div class="chat">
		<ul>
			<li v-for="message in messages">{{ message }}</li>
		</ul>
	</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/2.3.0/socket.io.js"></script>
<script src="https://code.jquery.com/jquery-1.11.1.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/vue/2.6.10/vue.js"></script>
</body>
<script>
    var socket = io();

    socket.on('reply', function (d) {
        console.log(d);
    });

	new Vue({
		el: '.chat',
		data: {
		    messages: []
		},
		mounted: function () {
            socket.on('reply', function (d) {
				this.messages.push(d)
            }.bind(this));
        }
	});

	console.log("msg");
    var s2 = io("/chat");
</script>
</html>
```

I've checked using nodeJS socket server version, and it works perfectly. Only the go socket version has this bug
```javascript
var app = require('express')();
var http = require('http').Server(app);
var io = require('socket.io')(http);

app.get('/', function (request, response) {
    response.sendFile(__dirname + '/index.html');
});

io.on('connection', function (socket) {
    console.log("new user");
    socket.emit('reply', 'Connected')
});

http.listen(3001);
```

You can verify this using the example given with this repo, just remove the line that creates another socket: 

```javascript
var s2 = io("/chat");
```

Just remove this line.

### PR DESCRIPTION
**Pull request by neilpa**
This should actually resolve the slowness reported in #312 by running `go mod tidy` to properly fix-up the dependencies. The attempt in #314 wasn't enough since [both versions were still in the `go.sum`](https://github.com/googollee/go-socket.io/pull/314/files#diff-f949e2d81c8076ebbf8af38fcbb72c1fL40-R43).

I've got a min-repro at [neilpa/go-socket.io-312](https://github.com/neilpa/go-socket.io-312/) that shows the 20 seconds vs <1ms difference below. The [`fork` branch updates the `go.mod`](https://github.com/neilpa/go-socket.io-312/compare/fork) to reference the changes in this PR.

**Before**

```
~ % git clone git@github.com:neilpa/go-socket.io-312
~ % cd go-socket.io-312
~/go-socket.io-312 % make
go build -o server
~/go-socket.io-312 % ./server& ./client.js
[1] 1515
server: 00:22:58.918432 Listening on localhost:5432
client: 00:22:59.008Z connecting to http://localhost:5432
server: 00:22:59.032414 connected: 1
client: 00:22:59.036Z sending foo: hello
server: 00:22:59.037426 recv'd foo, sending bar: hello
client: 00:23:19.037Z recv'd bar: hello
^C
~/go-socket.io-312 % kill 1515
[1]  + terminated  ./server
```

**After**
```
~/go-socket.io-312 % git co fork
Switched to branch 'fork'
Your branch is up to date with 'origin/fork'.
~/go-socket.io-312 % make
go build -o server
~/go-socket.io-312 % ./server& ./client.js
[1] 1673
server: 00:23:44.687166 Listening on localhost:5432
client: 00:23:44.750Z connecting to http://localhost:5432
server: 00:23:44.775161 connected: 1
client: 00:23:44.778Z sending foo: hello
server: 00:23:44.780298 recv'd foo, sending bar: hello
client: 00:23:44.780Z recv'd bar: hello
^C
```

### CODE REVIEW

**Comment by erkie** [comment]
@neilpa cool! Can you rebase and run again?

**Comment by neilpa** [comment]
@erkie Should be good now

**Comment by erkie** [comment]
Hmm... Looking at the dependencies it seems to add. It's adding a lot of dependencies that are only present in the `examples/` folder. I guess these should not be included in the `go.mod` since they are not needed for the project to run.

It seems like calling `go mod tidy`, there is no way to tell the tool to exclude the `example` directory. However, according to this thread: https://github.com/golang/go/issues/30058 if we rename the `example` directory to `_example`(leading underscore) the go tools will ignore it. I checked out your branch locally and got this diff:

```diff
diff --git a/go.mod b/go.mod
index 0c18aed..6bc2caf 100644
--- a/go.mod
+++ b/go.mod
@@ -3,23 +3,8 @@ module github.com/googollee/go-socket.io
 go 1.13

 require (
-       github.com/ajg/form v1.5.1 // indirect
-       github.com/fasthttp-contrib/websocket v0.0.0-20160511215533-1f3b11f56072 // indirect
-       github.com/gin-gonic/gin v1.5.0
        github.com/googollee/go-engine.io v1.4.2
-       github.com/imkira/go-interpol v1.1.0 // indirect
-       github.com/k0kubun/colorstring v0.0.0-20150214042306-9440f1994b88 // indirect
-       github.com/kataras/iris/v12 v12.1.8
-       github.com/mattn/go-colorable v0.1.4 // indirect
-       github.com/moul/http2curl v1.0.0 // indirect
-       github.com/onsi/ginkgo v1.12.0 // indirect
-       github.com/onsi/gomega v1.9.0 // indirect
-       github.com/smartystreets/goconvey v1.6.4 // indirect
+       github.com/gorilla/websocket v1.4.1 // indirect
        github.com/stretchr/testify v1.4.0
-       github.com/valyala/fasthttp v1.9.0 // indirect
-       github.com/xeipuuv/gojsonschema v1.2.0 // indirect
-       github.com/yalp/jsonpath v0.0.0-20180802001716-5cc68e5049a0 // indirect
-       github.com/yudai/gojsondiff v1.0.0 // indirect
-       github.com/yudai/golcs v0.0.0-20170316035057-ecda9a501e82 // indirect
-       github.com/yudai/pp v2.0.1+incompatible // indirect
+       gopkg.in/yaml.v2 v2.2.4 // indirect
 )
diff --git a/go.sum b/go.sum
```

Which is a lot cleaner. Can you do that in a commit and run `go mod tidy` again, and I'll merge this?

Thanks!!!!

**Comment by neilpa** [comment]
That's good to know. I've been a following a few of the other go module threads related to the testing deps but hadn't seen that one.

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/googollee/go-socket.io
 go 1.13
 
 require (
-	github.com/googollee/go-engine.io v1.4.3-0.20200220091802-9b2ab104b298
+	github.com/googollee/go-engine.io v1.4.2
```
Comment (erkie) [inline go.mod]: I feel bad for making so many comments on this small (but very great!) pull request. It looks like this downgrades the dependency here. Is that intentional?

Comment (neilpa) [reply go.mod]: No, but there isn't a `v1.4.3` tag on the go-engine.io repo. Looks like `v1.4.3` tag for this repo was [just created yesterday](https://github.com/googollee/go-socket.io/releases/tag/v1.4.3). I suspect that `go mod tidy` is just picking up the latest tag.

If the goal is to keep the versions in sync, I _think_ the thing to do is tag both a `v1.4.3` and `v1.4.4` release on go-engine.io repo. Those could be both point to the same commit.

Once that's done, bump the version in this PR, merge it, then tag a `v1.4.4` of go-socket.io.

**Comment by erkie** [comment]
I merged this now. The minor version thingie we can solve separately. Thanks again!!!

**Pull request merged**